pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/MSelectionOnlyOperator.java

```diff
@@ -47,24 +48,21 @@
 
   private final IndexSegment _indexSegment;
   private final MProjectionOperator _projectionOperator;
-  private final Selection _selection;
   private final DataSchema _dataSchema;
   private final Block[] _blocks;
-  private final String[] _selectionColumns;
   private final int _limitDocs;
   private final Collection<Serializable[]> _rowEvents;
   private ExecutionStatistics _executionStatistics;
 
   public MSelectionOnlyOperator(IndexSegment indexSegment, Selection selection, Operator projectionOperator) {
     _indexSegment = indexSegment;
-    _selection = selection;
-    _limitDocs = _selection.getSize();
+    _limitDocs = selection.getSize();
     _projectionOperator = (MProjectionOperator) projectionOperator;
-
-    _selectionColumns = SelectionOperatorUtils.extractSelectionRelatedColumns(_selection, _indexSegment);
-    _dataSchema = SelectionOperatorUtils.extractDataSchema(_selectionColumns, indexSegment);
-    _blocks = new Block[_selectionColumns.length];
-    _rowEvents = new ArrayList<Serializable[]>();
+    List<String> selectionColumns =
+        SelectionOperatorUtils.getSelectionColumns(selection.getSelectionColumns(), indexSegment);
```

Comment:
Not required for this PR, but we should have this operator class work off of projection operator only (ie indexSegment should not be passed to it).